sootup.core/src/main/java/sootup/core/graph/StmtGraph.java

```diff
@@ -203,7 +207,7 @@ public void validateStmtConnectionsInGraph() {
 
           for (Stmt target : successors) {
             if (target == stmt) {
-              throw new IllegalStateException(stmt + ": a Stmt cannot branch to itself.");
+              logger.warn(stmt + " Stmt contains a loop or Stmt branches to itself");
             }
```

Comment:
as this is valid situation created from valid bytecode I think we can even remove that warning